intake_stac/catalog.py

```diff
@@ -347,67 +323,71 @@ def stack_bands(self, bands):
                     f'{band} not found in list of eo:bands in collection.'
                     f'Valid values: {sorted(list(set(valid_band_names)))}'
                 )
-
-            value = assets.get(band)
-            band_type = value.get('type')
-            types.append(band_type)
-
-            href = value.get('href')
-            pattern = href.replace(band, '{band}')
-            if 'path_as_pattern' not in item:
-                item['path_as_pattern'] = pattern
-            elif item['path_as_pattern'] != pattern:
-                raise ValueError(
-                    f'Stacking failed: {href} does not contain '
-                    'band info in a fixed section of the url'
-                )
-
+            asset = assets.get(band)
             titles.append(band)
-            item['urlpath'].append(href)
+            types.append(asset.media_type)
+            hrefs.append(asset.href)
+            # metadata.update(asset.to_dict())
 
         unique_types = set(types)
         if len(unique_types) != 1:
             raise ValueError(
                 f'Stacking failed: bands must have type, multiple found: {unique_types}'
             )
-        else:
-            item['type'] = types[0]
 
-        item['title'] = ', '.join(titles)
-        return StacEntry('_'.join(bands), item, stacked=True)
+        configDict['name'] = '_'.join(bands)
+        configDict['description'] = ', '.join(titles)
+        configDict['args'] = dict(
+            chunks={}, concat_dim='band', path_as_pattern=path_as_pattern, urlpath=hrefs
+        )
+        configDict['metadata'] = {}  # blank for now, make union of assets ?
 
+        return CombinedAssets(configDict)
 
-class StacEntry(LocalCatalogEntry):
+
+class StacAsset(LocalCatalogEntry):
```

Comment:
renamed to match convention of other classes (StacAsset is what's being mapped to Intake Entry, as StacItem maps to Intake Catalog).